pkg/cloudprovider/aws/apis/v1alpha1/provider.go

```diff
@@ -49,6 +49,66 @@ type AWS struct {
 	// Tags to be applied on ec2 resources like instances and launch templates.
 	// +optional
 	Tags map[string]string `json:"tags,omitempty"`
+	// MetadataOptions for the generated launch template of provisioned nodes.
+	//
+	// This specifies the exposure of the Instance Metadata Service to
+	// provisioned EC2 nodes. For more information,
+	// see Instance Metadata and User Data
+	// (https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/ec2-instance-metadata.html)
+	// in the Amazon Elastic Compute Cloud User Guide.
+	//
+	// Refer to recommended, security best practices
+	// (https://aws.github.io/aws-eks-best-practices/security/docs/iam/#restrict-access-to-the-instance-profile-assigned-to-the-worker-node)
+	// for limiting exposure of Instance Metadata and User Data to pods.
+	// If omitted, defaults to httpEndpoint enabled, with httpProtocolIpv6
+	// disabled, with httpPutResponseLimit of 2, and with httpTokens
+	// required.
+	// +optional
+	MetadataOptions *MetadataOptions `json:"metadataOptions,omitempty"`
+}
+
+// MetadataOptions contains parameters for specifying the exposure of the
+// Instance Metadata Service to provisioned EC2 nodes.
+type MetadataOptions struct {
+	// HTTPEndpoint enables or disables the HTTP metadata endpoint on provisioned
+	// nodes. If metadata options is non-nil, but this parameter is not specified,
+	// the default state is "enabled".
+	//
+	// If you specify a value of "disabled", instance metadata will not be accessible
+	// on the node.
+	// +optional
+	HTTPEndpoint *string `json:"httpEndpoint,omitempty"`
+
+	// HTTPProtocolIpv6 enables or disables the IPv6 endpoint for the instance metadata
+	// service on provisioned nodes. If metadata options is non-nil, but this parameter
+	// is not specified, the default state is "disabled".
+	// +optional
+	HTTPProtocolIpv6 *string `json:"httpProtocolIpv6,omitempty"`
```

Comment:
Can you try to find an example of a K8s API that uses IPV6 vs Ipv6 vs IPv6 as a YAML **key**?

Reply:
On Node [`status.address.addressType`](https://kubernetes.io/docs/reference/kubernetes-api/cluster-resources/node-v1/#NodeStatus), `ExternalIP` and `InternalIP` are possible values. 

On Service, we have [`spec`](https://kubernetes.io/docs/reference/kubernetes-api/service-resources/service-v1/#ServiceSpec) fields like `clusterIP`, `clusterIPs`, `externalIPs`, etc. [Valid values for `ipFamilies`](https://kubernetes.io/docs/tasks/network/validate-dual-stack/) seems to include `IPv6`.

I'll go ahead and change this field to `httpProtocolIPv6`.

Reply:
I think I come to the same confusion. For further nuance, values follow UpperCamel where keys follow lowerCamel. I *think* because the `spec.clusterIP` uses caps, it means we should use `IPv6` instead of `Ipv6`